api/src/main/java/com/example/api/service/ranking/RankingService.java

```diff
@@ -0,0 +1,103 @@
+package com.example.api.service.ranking;
+
+import com.example.api.dto.response.ranking.RankingEntry;
+import com.example.api.model.user.AccountType;
+import com.example.api.model.user.User;
+import com.example.api.repo.activity.result.FileTaskResultRepo;
+import com.example.api.repo.activity.result.GraphTaskResultRepo;
+import com.example.api.repo.user.UserRepo;
+import com.example.api.service.activity.feedback.ProfessorFeedbackService;
+import com.example.api.service.activity.result.GraphTaskResultService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+@Transactional
+public class RankingService {
+    private final UserRepo userRepo;
+    private final GraphTaskResultRepo graphTaskResultRepo;
+    private final FileTaskResultRepo fileTaskResultRepo;
+    private final GraphTaskResultService graphTaskResultService;
+    private final ProfessorFeedbackService professorFeedbackService;
+
+    public List<RankingEntry> getRanking() {
+        return userRepo.findAllByAccountTypeEquals(AccountType.STUDENT)
```

Comment:
To mi nie działa jak odpaliłem lokalnie, w sensie wypisuje studentów, ale jak dodałem jakies punkty do resultów, to i tak każdy student ma 0.

Reply:
Niżej napisze gdzie wydaje mi się, że jest bląd.

Reply:
po zmianach u mnie lokalnie działa